apps/sync/js/adapters/history.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global IAC,
+  SyncEngine
+*/
+
+var HistoryHelper = {
+  set syncedCollectionModifiedTime(mtime) {
+    window.asyncStorage.setItem('LastSyncedStatus::Collection::mtime', mtime);
+  },
+  get syncedCollectionModifiedTime() {
+    return new Promise((resolve, reject) => {
+      window.asyncStorage.
+      getItem('LastSyncedStatus::Collection::mtime', (mtime) => {
+        resolve(mtime);
+      });
+    });
+  },
+
+  addPlaces(places) {
+    return IAC.request('sync-history', {
+      method: 'addPlaces',
+      args: [places]
+    });
+  },
+
+  addPlace(place) {
+    return IAC.request('sync-history', {
+      method: 'addPlace',
+      args: [place]
+    });
+  }
+};
+
+SyncEngine.DataAdapterClasses.history = {
+  _fullSync(kintoCollection, lastModifiedTime) {
+    function updateHistoryCollection(list) {
+      var historyRecords = list.data;
+      var places = [];
+      var i;
+      for (i = 0; i < historyRecords.length; i++) {
+        if (historyRecords[i].last_modified <= lastModifiedTime) {
```

Comment:
I still would like to see https://github.com/mozilla-b2g/gaia/pull/31896/files#r39845952 :) It saves some memory.